extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java

```diff
@@ -68,4 +97,110 @@ public String getRelatedUrl() {
+        }
+
+        throw new InvalidSourceException("Unable to find a valid subscriptions.csv file (try extracting and selecting the csv file)");
+    }
+
+    public List<SubscriptionItem> fromCsvInputStream(@Nonnull final InputStream contentInputStream)
+            throws ExtractionException {
+        // Expected format of CSV file:
+        //      Channel Id,Channel Url,Channel Title
+        //      UC1JTQBa5QxZCpXrFSkMxmPw,http://www.youtube.com/channel/UC1JTQBa5QxZCpXrFSkMxmPw,Raycevick
+        //      UCFl7yKfcRcFmIUbKeCA-SJQ,http://www.youtube.com/channel/UCFl7yKfcRcFmIUbKeCA-SJQ,Joji
+        //
+        // Notes:
+        //      It's always 3 columns
+        //      The first line is always a header
+        //      Header names are different based on the locale
+        //      Fortunately the data is always the same order no matter what locale
+
+        int currentLine = 0;
+        String line = "";
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(contentInputStream))) {
+            final List<SubscriptionItem> subscriptionItems = new ArrayList<>();
+
+            // Ignore header
+            currentLine = 1;
+            line = br.readLine();
+
+            while ((line = br.readLine()) != null) {
+                currentLine++;
+
+                // Exit early if we've read the first few lines and we haven't added any items
+                // It's likely we're in the wrong file
+                if (currentLine > 5 && subscriptionItems.size() == 0) {
+                    break;
+                }
+
+                // First comma
+                int i1 = line.indexOf(",");
+                if (i1 == -1) {
+                    continue;
+                }
+
+                // Second comma
+                int i2 = line.indexOf(",", i1 + 1);
+                if (i2 == -1) {
+                    continue;
+                }
+
+                // Third comma or line length
+                int i3 = line.indexOf(",", i2 + 1);
+                if (i3 == -1) {
+                    i3 = line.length();
+                }
+
+                // Channel URL from second entry
+                final String channelUrl = line
+                        .substring(i1 + 1, i2)
+                        .replace("http://", "https://");
+                if (!channelUrl.startsWith(BASE_CHANNEL_URL)) {
+                    continue;
+                }
+
+                // Channel title from third entry
+                final String channelTitle = line.substring(i2 + 1, i3);
+
+                final SubscriptionItem newItem = new SubscriptionItem(service.getServiceId(), channelUrl, channelTitle);
+                subscriptionItems.add(newItem);
+            }
+
+            return subscriptionItems;
+        } catch (IOException e) {
+            if (line == null) {
+                line = "<null>";
+            }
+            if (line.length() > 10) {
```

Comment:
```suggestion
            else if (line.length() > 10) {
```

if the line is ``null`` the check for the length > 10 is irrelevant